content/hardware/04.pro/boards/portenta-x8/tutorials/image-building/content.md

```diff
@@ -0,0 +1,211 @@
+
+Follow the instructions, after which you should have two new files in the directory: `<yourSSHkey> and <yourSSHkey>.pub`
+
+Now initialize the `ssh-agent` with `eval "$(ssh-agent -s)"`. Add your key with `ssh-add ~/.ssh/<yourSSHkey>`.
+
+***Make sure your `ssh-agent` is running, you can check the keys that your agent has with the command `ssh-add -l` if there is no key attached repeat the steps of adding the key and check if it was successfully added***
+
+### Foundries.io API Token
+
+***This step is only mandatory if you want to get the source code from your Factory***
+
+To get the needed files (source code) for compiling the image we need access to our Factory repository.
+
+To make that happen, go to your [Foundries account > Settings > Api Tokens](https://app.foundries.io/settings/tokens/):
+![API token page on Foundries.io](assets/foundries_API_tokens.png)
+
+Press the "New Token" button and follow the instructions.
+![API token creation page](assets/foundries_API_token_create.png)
+
+Now make sure you select the correct scopes to "Use for source code access" and choose your derived Factory.
+![API token creation page](assets/foundries_API_token_create_scopes.png)
+
+Once it has been generated, copy the token shown.
+![API token creation page](assets/foundries_API_token_created.png)
+
+On your machine create a variable to store the token and add it to the global configuration for Git:
+
+```
+API_TOKEN="<YourToken>"
+git config --global http.https://source.foundries.io.extraheader "Authorization: basic $(echo -n $API_TOKEN | openssl base64)"
+```
+
+Check that your token is properly configured by cloning the "containers" repository of your Factory:
+
+```
+git clone https://source.foundries.io/factories/<factory>/containers.git
+```
+
+Make sure you clone the repository outside our `builds` folder. You can remove it after you finish your check.
+
+You can check your Git configuration with `git config --global -l` you should see:
+
+```
+http.https://source.foundries.io.extraheader=Authorization: basic <yourToken on base64>
+```
+
+### Google Repo: Fetch Yocto Project Layers
+
+Google's Repo application handles fetching the repositories (Yocto Project layers) that will be used in the build process.
+
+Use these commands to:
+
+1. create a `.bin` folder to install the `repo` application in
+2. add it to the system `PATH` so you will be able to use the `repo` command anywhere
+3. download repo into `.bin`
+4. set permission so that all users and run it.
+
+```
+mkdir -p ~/builds/.bin
+PATH="${HOME}/builds/.bin:${PATH}"
+curl https://storage.googleapis.com/git-repo-downloads/repo > ~/builds/.bin/repo
+chmod a+rx ~/builds/.bin/repo
+```
+
+Now you should be able to access the command `repo` anywhere.
+
+### Set up Your Repository
+
+Switch to your build directory and create a folder to store the source code:
+
+```
+cd ~/builds/
+mkdir portenta-x8
+cd portenta-x8
+```
+
+Get your repository link:
```

Comment:
```suggestion
Get your repository link. If you do not have a FoundriesFactory subscription, use the Public repository link, otherwise you can use your Factory repository link:
```